File: Java/Tema7_Ejercicio2.java
/*
 * TEMA 7: FUNCIONES Y MÉTODOS - EJERCICIO 2
 * ==========================================
 * Escriba un programa con funciones para operaciones matemáticas básicas.
 */

public class Tema7_Ejercicio2 {
    
    static double suma(double a, double b) {
        return a + b;
    }
    
    static double resta(double a, double b) {
        return a - b;
    }
    
    static double multiplicacion(double a, double b) {
        return a * b;
    }
    
    static double division(double a, double b) {
        if (b == 0) {
            System.out.println("Error: No se puede dividir entre cero.");
            return 0;
        }
        return a / b;
    }
    
    public static void main(String[] args) {
        double num1 = 20;
        double num2 = 5;
        
        System.out.println("=== OPERACIONES MATEMÁTICAS ===");
        System.out.println(num1 + " + " + num2 + " = " + suma(num1, num2));
        System.out.println(num1 + " - " + num2 + " = " + resta(num1, num2));
        System.out.println(num1 + " × " + num2 + " = " + multiplicacion(num1, num2));
        System.out.println(num1 + " ÷ " + num2 + " = " + division(num1, num2));
    }
}

/*
 * SALIDA ESPERADA:
 * === OPERACIONES MATEMÁTICAS ===
 * 20.0 + 5.0 = 25.0
 * 20.0 - 5.0 = 15.0
 * 20.0 × 5.0 = 100.0
 * 20.0 ÷ 5.0 = 4.0
 */

File: JS/Tema7_Ejercicio2.js
/*
 * TEMA 7: FUNCIONES Y MÉTODOS - EJERCICIO 2
 * =========================================
 * Funciones de utilidad matemática.
 * 
 * EXPLICACIÓN:
 * - Funciones para operaciones comunes
 * - Métodos Math
 * - Validación de entrada
 */

function absoluto(n) {
    return Math.abs(n);
}

function raiz(n) {
    return n >= 0 ? Math.sqrt(n) : "Error: número negativo";
}

function redondear(n, decimales = 0) {
    return Math.round(n * Math.pow(10, decimales)) / Math.pow(10, decimales);
}

function esNumero(valor) {
    return !isNaN(valor) && isFinite(valor);
}

console.log("=== FUNCIONES DE UTILIDAD ===");

console.log(`Valor absoluto:`);
console.log(`|−15| = ${absoluto(-15)}`);
console.log(`|42| = ${absoluto(42)}`);

console.log(`\nRaíz cuadrada:`);
console.log(`√16 = ${raiz(16)}`);
console.log(`√-9 = ${raiz(-9)}`);

console.log(`\nRedondeo:`);
console.log(`3.14159 a 2 decimales = ${redondear(3.14159, 2)}`);
console.log(`3.14159 a 0 decimales = ${redondear(3.14159)}`);

console.log(`\nValidación:`);
console.log(`¿42 es número? ${esNumero(42)}`);
console.log(`¿"texto" es número? ${esNumero("texto")}`);

/*
 * SALIDA ESPERADA:
 * === FUNCIONES DE UTILIDAD ===
 * Valor absoluto:
 * |−15| = 15
 * |42| = 42
 * 
 * Raíz cuadrada:
 * √16 = 4
 * √-9 = Error: número negativo
 * 
 * Redondeo:
 * 3.14159 a 2 decimales = 3.14
 * 3.14159 a 0 decimales = 3
 * 
 * Validación:
 * ¿42 es número? true
 * ¿"texto" es número? false
 */
